xml/src/main/java/org/ehcache/xml/ConfigurationParser.java

```diff
@@ -89,16 +91,21 @@
 
   private static final SchemaFactory XSD_SCHEMA_FACTORY = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
 
+  private static final URL CORE_SCHEMA_URL = XmlConfiguration.class.getResource("/ehcache-core.xsd");
+  private static final String CORE_SCHEMA_NAMESPACE = "http://www.ehcache.org/v3";
+  private static final String CORE_SCHEMA_ROOT_ELEMENT = "config";
+  private static final String CORE_SCHEMA_JAXB_MODEL_PACKAGE = "org.ehcache.xml.model";
+
   private final Map<URI, CacheManagerServiceConfigurationParser<?>> xmlParsers = new HashMap<URI, CacheManagerServiceConfigurationParser<?>>();
   private final Map<URI, CacheServiceConfigurationParser<?>> cacheXmlParsers = new HashMap<URI, CacheServiceConfigurationParser<?>>();
   private final Unmarshaller unmarshaller;
+  private final Map<URI, CacheResourceConfigurationParser> resourceXmlParsers = new HashMap<URI, CacheResourceConfigurationParser>();
   private final ConfigType config;
 
-  public ConfigurationParser(String xml, URL... sources) throws IOException, SAXException, JAXBException, ParserConfigurationException {
+  public ConfigurationParser(String xml) throws IOException, SAXException, JAXBException, ParserConfigurationException {
     Collection<Source> schemaSources = new ArrayList<Source>();
-    for (URL source : sources) {
-      schemaSources.add(new StreamSource(source.openStream()));
-    }
+    schemaSources.add(new StreamSource(CORE_SCHEMA_URL.openStream()));
+
     for (CacheManagerServiceConfigurationParser<?> parser : ClassLoading.libraryServiceLoaderFor(CacheManagerServiceConfigurationParser.class)) {
```

Comment:
Looking at this logic, I wonder if we want to file an issue for turning these in to a tree.  Top level service that provides the schema, followed by three accessor to get at the (up to) three parsers?